### PR TITLE
Misc fixes

### PR DESCRIPTION
**Pull request by dfordivam**
Contains #10 
And these fixes on top of it

- Rename jsaddle-core.js function jsaddle -> jsaddleCoreJs
- Add runJSMCheap, which does not do catchError
- make unsafeInlineLiftIO work in MonadJSM
- Create async callback with newAsyncCallback
- Use separate syncState MVar for new thread
- Use JavaScriptException in MonadError
- Remove JSException and Exception module

### CODE REVIEW

File: jsaddle/js/jsaddle-core.js [hunk 1]
```diff
@@ -176,33 +176,54 @@ function jsaddle(global, sendRsp, processSyncCommand, RESPONSE_BUFFER_MAX_SIZE)
...
-      var req = syncRequests.dequeue();
-      if(!req.Right) {
-        throw "processAllEnqueuedReqs: req is not Right; this should never happen because Lefts should only be sent while a synchronous request is still in progress";
+      var tuple = syncRequests.dequeue();
+      var syncReq = tuple[1];
+      if(syncReq.tag !== 'Req') {
+        throw "processAllEnqueuedReqs: syncReq is not SyncBlockReq_Req; this should never happen because Result/Throw should only be sent while a synchronous request is still in progress";
+      }
+      if (tuple[0] > syncDepth) {
+        console.warn ("processAllEnqueuedReqs: queue contains a request for a frame which has exited");
```
Comment (matthewbauer) [inline jsaddle/js/jsaddle-core.js]: Did you consider making this fatal? We can keep going here, but am I correct in thinking something is really wrong if we get the wrong syncDepth?

File: jsaddle/src/Language/Javascript/JSaddle/Types.hs [hunk 2]
```diff
@@ -532,6 +556,18 @@ instance ToJSON SyncCommand where
 instance FromJSON SyncCommand where
   parseJSON = A.genericParseJSON $ aesonOptions "SyncCommand"
 
+data SyncBlockReq
+  = SyncBlockReq_Req TryReq
+  | SyncBlockReq_Result ValId
+  | SyncBlockReq_Throw Int Text
```
Comment (matthewbauer) [inline jsaddle/src/Language/Javascript/JSaddle/Types.hs]: Could you add a doc for Int here. It should be the sync depth the throw was sent at right?

File: jsaddle/src/Language/Javascript/JSaddle/Types.hs [hunk 3]
```diff
@@ -522,7 +546,7 @@ instance FromJSON Rsp where
   parseJSON = A.genericParseJSON $ aesonOptions "Rsp"
 
 data SyncCommand
-   = SyncCommand_StartCallback CallbackId ValId [ValId] -- The input valIds here must always be allocated on the JS side --TODO: Make sure throwing stuff works when it ends up skipping over our own call stack entries
+   = SyncCommand_StartCallback Bool CallbackId ValId [ValId] -- The input valIds here must always be allocated on the JS side --TODO: Make sure throwing stuff works when it ends up skipping over our own call stack entries
```
Comment (matthewbauer) [inline jsaddle/src/Language/Javascript/JSaddle/Types.hs]: Also add doc for Bool here

File: jsaddle/js/jsaddle-core.js (hunk 1 above)
Comment (dfordivam) [reply jsaddle/js/jsaddle-core.js]: fixed this in #10

File: jsaddle/src/Language/Javascript/JSaddle/Types.hs (hunk 2 above)
Comment (dfordivam) [reply jsaddle/src/Language/Javascript/JSaddle/Types.hs]: Done

File: jsaddle/src/Language/Javascript/JSaddle/Types.hs (hunk 3 above)
Comment (dfordivam) [reply jsaddle/src/Language/Javascript/JSaddle/Types.hs]: done

**Review by matthewbauer — APPROVED** [review]

**Pull request merged**